src/main.js

```diff
@@ -1,71 +1,85 @@
-import { createFilmCardTemplate } from './view/film-card-view';
-import { createFilmsCatalogTemplate } from './view/films-catalog-view';
-import { createFooterStatisticsTemplate } from './view/footer-statistics';
-import { createMenuTemplate } from './view/menu-view';
-import { createPopupTemplate } from './view/popup-view';
-import { createProfileRatingTemplate } from './view/profile-view';
-import { createShowMoreButtonTemplate } from './view/show-more-button-view';
-import { createSortingTemplate } from './view/sorting-view';
-import { createFilmCommentTemplate } from './view/comments-view';
 import { generateMovieData, currentComments } from './mock/task.js';
 import { compareCommentsAmount, compareMoviesRating } from './util.js';
+import { RenderPosition, render } from './render.js';
+import PopupView from './view/popup-view';
+import ProfileRatingView from './view/profile-view';
+import MenuView from './view/menu-view';
+import FilmCardTemplateView from './view/film-card-view';
+import SortingView from './view/sorting-view';
+import ShowMoreButtonView from './view/show-more-button-view';
+import FooterStatisticsView from './view/footer-statistics';
+import FilmCommentView from './view/comments-view';
+import FilmsCatalogView from './view/films-catalog-view';
+
+const bodyELement = document.querySelector('body');
+const headerElement = document.querySelector('.header');
+const mainElement = document.querySelector('.main');
+const footerStatisticElement = document.querySelector('.footer__statistics');
+const footerElement = document.querySelector('.footer');
 
 const FILMS_PER_STEP = 5;
 const MOVIE_DATA_OBJECTS = 20;
 const TOP_MOVIES_TO_SHOW = 2;
 
+
 const movies = Array.from({length: MOVIE_DATA_OBJECTS}, generateMovieData);
 
 const watchlistMovies = movies.filter((movie) => movie.user_details.watchlist === true);
 const alreadyWatchedMovies = movies.filter((movie) => movie.user_details.already_watched === true);
 const favoriteMovies = movies.filter((movie) => movie.user_details.favorite === true);
 
 const topRatedMovies = movies.slice().sort(compareMoviesRating);
-
 const mostCommentedMovies = movies.slice().sort(compareCommentsAmount);
 
-const RenderPosition = {
-  BEFOREBEGIN: 'beforebegin',
-  AFTERBEGIN: 'afterbegin',
-  BEFOREEND: 'beforeend',
-  AFTEREND: 'afterend',
-};
 
-const renderTemplate = (container, template, place) => {
-  container.insertAdjacentHTML(place, template);
-};
+render(headerElement, new ProfileRatingView(alreadyWatchedMovies.length).element, RenderPosition.BEFOREEND);
```

Comment:
создание компонентов лучше выносить в переменные, они могут нам в дальнейшем понадомиться и нам не придётся пересоздавать их (DRY)
```suggestion
const profileRatingView = new ProfileRatingView(alreadyWatchedMovies.length);
render(headerElement, profileRatingView .element, RenderPosition.BEFOREEND);
```

Исправь все остальные создания компонентов по такому примеру.